src/test/java/io/jenkins/plugins/file_parameters/FileParameterWrapperTest.java

```diff
@@ -58,4 +62,153 @@
         r.assertLogContains("loaded 'UPLOADED CONTENT HERE' from ", b);
     }
 
+    @Test public void base64UndefinedFail() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+        p.addProperty(new ParametersDefinitionProperty(new Base64FileParameterDefinition("FILE", null)));
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File(name:'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter('FILE') {\n" +
+            "          echo('foo') \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        r.assertBuildStatus(Result.FAILURE, run);
+        r.assertLogContains("No parameter named FILE", run);
+    }
+
+    @Test public void base64WithAllowNoFile() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+        p.addProperty(new ParametersDefinitionProperty(new Base64FileParameterDefinition("FILE", null)));
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File(name:'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter(name:'FILE', allowNoFile: true) {\n" +
+            "          echo('foo') \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        r.assertBuildStatus(Result.SUCCESS, run);
+        r.assertLogContains("foo", run);
+        r.assertLogContains("Skip file parameter as there is no parameter with name: 'FILE'", run);
+    }
+
+    @Test public void base64DeclarativeParameterCreated() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File (name: 'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter('FILE') {\n" +
+            "          echo(/loaded '${readFile(FILE).toUpperCase(Locale.ROOT)}' from $FILE/) \n" +
```

Comment:
BTW while `sh 'cat $FILE'` is most realistic for docs, I prefer `readFile` for functional tests because it requires no special work to pass on Windows.